File: app/src/main/java/com/filemanager/fileexplorer/myfile/cleanmaster/free/Iinterface/CallbackFav.java
package com.filemanager.fileexplorer.myfile.cleanmaster.free.Iinterface;

import com.filemanager.fileexplorer.myfile.cleanmaster.free.DTO.File_DTO;

import java.util.ArrayList;


public class CallbackFav {
    private static CallbackFav mInstance;
    private ArrayList<File_DTO> file_dtos;
    private OnCustomStateListener mListener;


    public interface OnCustomStateListener {
        void stateChanged();
    }

    private CallbackFav() {
    }

    public static CallbackFav getInstance() {
        if (mInstance == null) {
            mInstance = new CallbackFav();
        }
        return mInstance;
    }

    public void setListener(OnCustomStateListener onCustomStateListener) {
        this.mListener = onCustomStateListener;
    }

    public void changeState(ArrayList<File_DTO> arrayList) {
        if (this.mListener != null) {
            this.file_dtos = arrayList;
            notifyStateChange();
        }
    }

    private void notifyStateChange() {
        this.mListener.stateChanged();
    }

    public ArrayList<File_DTO> getState() {
        return this.file_dtos;
    }
}

File: app/src/main/java/com/demo/filemanager/Iinterface/CallbackFav.kt
package com.demo.filemanager.Iinterface

import com.demo.filemanager.DTO.File_DTO

class CallbackFav private constructor() {
    var state: ArrayList<File_DTO>? = null
        private set
    private var mListener: OnCustomStateListener? = null

    interface OnCustomStateListener {
        fun stateChanged()
    }

    fun setListener(onCustomStateListener: OnCustomStateListener?) {
        mListener = onCustomStateListener
    }

    fun changeState(arrayList: ArrayList<File_DTO>?) {
        if (mListener != null) {
            state = arrayList
            notifyStateChange()
        }
    }

    private fun notifyStateChange() {
        mListener!!.stateChanged()
    }

    companion object {
        private var mInstance: CallbackFav? = null
        val instance: CallbackFav?
            get() {
                if (mInstance == null) {
                    mInstance = CallbackFav()
                }
                return mInstance
            }
    }
}
